Marlin/src/gcode/temp/M106_M107.cpp

```diff
@@ -98,6 +116,10 @@ void GcodeSuite::M106() {
  * M107: Fan Off
  */
 void GcodeSuite::M107() {
+  #if ENABLED(FAN_MULTIPLIER)
+    if (thermalManager.lock_fan) return;
```

Comment:
The exception should be if this is because the print finished, but I didn't find a way to add that exception